### PR TITLE
Separate Put and Patch Options

### PR DESCRIPTION
**Pull request by kevindelgado**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Removes a TODO to separate Put and Patch Options now that we're using PatchOptions now

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by alexzielenski** [comment]
Feels like this PR includes lots of changes to make `ServerSideFIeldValidation` GA. Is that intentional?

**Comment by kevindelgado** [comment]
> Feels like this PR includes lots of changes to make `ServerSideFIeldValidation` GA. Is that intentional?

This was stacked on https://github.com/kubernetes/kubernetes/pull/115271 Rebased now that that's merged

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go [hunk 1]
```diff
@@ -352,9 +352,10 @@ func (b *builder) buildRoute(root, path, httpMethod, actionVerb, operationVerb s
 		endpoints.AddObjectParams(b.ws, route, &metav1.GetOptions{})
 	case "list", "deletecollection":
 		endpoints.AddObjectParams(b.ws, route, &metav1.ListOptions{})
-	case "put", "patch":
-		// TODO: PatchOption added in feature branch but not in master yet
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:356]: while we're cleaning up TODOs, also drop the `// TODO: CRD support for export is still under consideration` above... export is removed

Comment (kevindelgado) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:356]: done

**Pull request merged**